Controller/FormController.php

```diff
@@ -80,18 +78,6 @@ protected function getTwoFactorToken(): TwoFactorTokenInterface
         return $token;
     }
 
-    protected function setPreferredProvider(Request $request, TwoFactorTokenInterface $token): void
```

Comment:
I'd like to keep this in the controller. I don't want to force the user to do it in a specific way, that's why its built into the controller in that extensible way.

Reply:
I'm not sure to understand about forcing user here. The logic is the same on the listener, what is the difference? :thinking:

Reply:
Ok it's a matter of class overwrite, right. I'm not sure how to deal with it, maybe make the listener more extensible would be a good compromise?

I'll look at https://github.com/scheb/two-factor-bundle/pull/226/files#r301256899 first, maybe I will not have to change anything thanks to that!